### PR TITLE
Added ProtBERT tutorial

### PR DESCRIPTION
**Pull request by Shiva-sankaran**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: examples/tutorials/Introduction_to_ProtBERT.ipynb
Comment (dhuvik) [inline examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>Shameless self plug, but also of functional interest: i think you can outsource some of the background to Elisa and I's Protein LM tutorial.</p><p> </p><p><a href="https://colab.research.google.com/drive/13eXPgZpzTOL3c_S7OM7uR6btP8ZWw7zn#scrollTo=YmeQ4fyq38_o" target="_blank">https://colab.research.google.com/drive/13eXPgZpzTOL3c_S7OM7uR6btP8ZWw7zn#scrollTo=YmeQ4fyq38_o</a></p><p> </p><p>And then transition directly into protBert in the context of DeepChem.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4041/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (dhuvik) [inline examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>is there any variation to the BERT architecture or pretraining objective? My understanding is that there isn't, just the pretraining data. Unclear given the use of adapts, if it is in fact different, Worth highlighting those differences.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4041/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (dhuvik) [inline examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>Perfect synopsis, well structured and brief.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4041/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (dhuvik) [inline examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>maybe show the dataset a little bit and even though its a dummy dataset just show and explain that the model should be getting this result. Especially if its trivial, lean into the motivation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4041/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

**Comment by NimishaDey** [comment]
LGTM

File: examples/tutorials/Introduction_to_ProtBERT.ipynb
Comment (Shiva-sankaran) [reply examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>I think that would be great. We should reference the general Protein LM tutorial from here once it has been merged into DeepChem.</p><p> </p>

Comment (Shiva-sankaran) [reply examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>Updated data from dummy dataset to DeepLoc dataset.</p>

Comment (Shiva-sankaran) [reply examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>Added pre-training strategy mentioned in the paper. "The authors first pretrain on protein sequences with lengths less than 512, then on sequences less than 1024, and finally on sequences up to 40,000."</p>

File: examples/tutorials/Introduction_to_ProtBERT.ipynb
Comment (davidRFB) [inline examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>Since the pretraining is a key part of the tutorial, can we see a plot of the training ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4041/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (davidRFB) [inline examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>It is possible to explain a bit more the meaning of this result ?  Is there  a comparison  with other method  ? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4041/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (Shiva-sankaran) [reply examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>The result showcases how well the current model predicts soluble vs non-soluble protein correctly. There are other works [1] [2]  that we can compare against but in this tutorial we are not training on the complete DeepLoc data since thats computationally its very expensive, so comparing results against other works might be unfair. However, to check if the implementation was correct I did run this notebook on a dedicated GPU server we were able to match the results published in ProtTrans with a margin of 5%, but it takes a few hours for the training to complete.</p><p> </p><p>[1] Heinzinger, Michael, et al. "Modeling aspects of the language of life through transfer-learning protein sequences." <em>BMC bioinformatics</em> 20 (2019): 1-17.</p><p>[2] Thumuluri, Vineet, et al. "DeepLoc 2.0: multi-label subcellular localization prediction using protein language models." <em>Nucleic acids research</em> 50.W1 (2022): W228-W234.</p>

Comment (Shiva-sankaran) [reply examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>Added.</p>

Comment (Shiva-sankaran) [reply examples/tutorials/Introduction_to_ProtBERT.ipynb:1]: <p>I have added some explanation regarding the final result.</p>

**Review by rbharath — COMMENTED** [review]
LGTM to merge once we add this file to the CSVs to etc for rendering.

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**
